src/js/components/pipelines/DataTable.jsx

```diff
@@ -49,7 +49,12 @@ var DataTable = React.createClass({
         schema = this.props.dataset.schema();
 
     this.hideFull(evt);
-    this.setState({fullField: schema[name]});
+    this.setState({
+      // fullField is used for rendering
+      fullField: schema[name],
+      // selectedField is used to pass data around while dragging
+      selectedField: schema[name]
+    });
```

Comment:
Separating these out fixes the issue where `hideFull` would null out the data pointer needed by the drop handler